client/components/site-selector/index.jsx

```diff
@@ -212,6 +212,10 @@ export default React.createClass( {
 		const recentSites = sites.map( function( site ) {
 			var siteHref;
 
+			if ( ! site ) {
```

Comment:
This will stick a null in the map. Do you want to chain a filter on there too to remove the null entries?